### PR TITLE
Release v4.0.1

### PR DESCRIPTION
**Pull request by Garandor**
Signed-off-by: Adam Reif <Garandor@manta.network>

## Description
partly adddresses #965

- MantaPay used to not follow crate version convention. Now it does.
- use manta_primitives::RocksDbWeight -> use manta_primitives::constants::RocksDbWeight
- Manta Runtime Benchmarks failing tracked in #995 

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: runtime/calamari/src/weights/pallet_parachain_staking.rs [hunk 1]
```diff
@@ -352,74 +353,74 @@ impl<T: frame_system::Config> pallet_parachain_staking::WeightInfo for Substrate
...
     // Storage: ParachainStaking DelayedPayouts (r:0 w:1)
     fn round_transition_on_initialize(x: u32, _y: u32, ) -> Weight {
-        (1_694_504_000 as Weight)
-            // Standard Error: 328_000
-            .saturating_add((2_649_000 as Weight).saturating_mul(x as Weight))
-            .saturating_add(T::DbWeight::get().reads(128 as Weight))
+        (1_507_092_000 as Weight)
+            // Standard Error: 360_000
+            .saturating_add((3_465_000 as Weight).saturating_mul(x as Weight))
+            .saturating_add(T::DbWeight::get().reads(119 as Weight))
```
Comment (ferrell-code) [inline runtime/calamari/src/weights/pallet_parachain_staking.rs:359]: Strange that the  comments above the benchmark don't update with the change in number of reads/writes.

Maybe hbs template changed

Edit: I think this changed a while ago not from this release

File: runtime/calamari/src/weights/pallet_tx_pause.rs [hunk 2]
```diff
@@ -54,13 +55,13 @@ pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_tx_pause::WeightInfo for SubstrateWeight<T> {
     // Storage: TransactionPause PausedTransactions (r:1 w:1)
     fn pause_transaction() -> Weight {
-        (18_286_000 as Weight)
+        (33_718_000 as Weight)
```
Comment (ghzlatarev) [inline runtime/calamari/src/weights/pallet_tx_pause.rs:58]: weird 2x

**Review by ghzlatarev — APPROVED** [review]

**Review by ferrell-code — APPROVED** [review]
Comment above, not sure if it is already a known thing.

File: runtime/calamari/src/weights/pallet_parachain_staking.rs (hunk 1 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_parachain_staking.rs:359]: Good catch, the hbs did not change, the comment is supposed to be output by benchmarking and there's a placeholder for it in https://github.com/Manta-Network/Manta/blob/release-v4.0.1/.github/resources/frame-weight-template.hbs#L53

Not sure why it doesn't actually work. Couldn't find any substrate issues on this on a quick search

File: runtime/calamari/src/weights/pallet_tx_pause.rs (hunk 2 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_tx_pause.rs:58]: doesn't seem to be due to code changes in the pallet, only reason coming to mind is the deps upgrade.

I'm really starting to think whatever that "bare metal" device is we're getting from AWS is not actually *bare* and whatever management software they're running is mucking up the benchmarks on the 1-50us time scale.
Unless we're running some tasks on it in parallel with benchmarking accidentally which could introduce all these fluctuations

Comment (Garandor) [reply runtime/calamari/src/weights/pallet_tx_pause.rs:58]: There's also the - more likely - possibility that benchmarking just sucks on that timescale.

Our billion long private extrinsics have differences between versions on the order of 100mil weight.
100mil = 100us . Non-realtime OSes have trouble reproducibly computing anything around a microsecond timescale (drivers, networking, scheduling) and combined with the AWS stuff above I'm starting to lean towards dismissing any benchmarking diffs below ~50us out of hand as benchmarking artifacts.

In our context of few large TXNs per block these small differences will barely matter anyway

File: runtime/calamari/src/weights/pallet_parachain_staking.rs (hunk 1 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_parachain_staking.rs:359]: Resolving as the comment isn't terribly important and I expect this to get autofixed with some of the next deps upgrades

File: runtime/calamari/src/weights/pallet_tx_pause.rs (hunk 2 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_tx_pause.rs:58]: Rerunning benchmark twice to verify
- https://github.com/Manta-Network/Manta/actions/runs/3955495077
- https://github.com/Manta-Network/Manta/actions/runs/3955501543

**Comment by Garandor** [comment]
Leaving this open for tonight, @ghzlatarev merge when you think acceptable

**Review by Dengjianping — APPROVED** [review]

File: runtime/calamari/src/weights/pallet_tx_pause.rs (hunk 2 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_tx_pause.rs:58]: Both benchmarks got cancelled. Merging anyway

**Pull request merged**